app/src/main/java/agency/five/codebase/android/movieapp/ui/component/MovieCategoryLabel.kt

```diff
@@ -0,0 +1,68 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import androidx.annotation.StringRes
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.Divider
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+sealed class MovieCategoryLabelTextViewState();
+
+data class MovieCategoryTextString(val category: String) : MovieCategoryLabelTextViewState()
+data class MovieCategoryTextStringResource(val category: Int) : MovieCategoryLabelTextViewState()
+
+data class MovieCategoryLabelViewState(
+    val itemId: Int,
+    val categoryText: MovieCategoryLabelTextViewState,
+)
+
+@Composable
+fun MovieCategoryLabel(
+    item: MovieCategoryLabelViewState,
+    state: MutableState<Boolean>,
+    onClick: () -> Unit,
+    modifier: Modifier
+) {
+    Box(
+        modifier = modifier
+            .clickable { onClick.invoke() }
+    ) {
+        Text(
+            text = when (item.categoryText) {
+                is MovieCategoryTextString -> item.categoryText.category;
+                is MovieCategoryTextStringResource -> stringResource(id = item.categoryText.category)
+            },
+            fontSize = 16.sp,
+            textDecoration = if (state.value) TextDecoration.Underline else TextDecoration.None,
+            fontWeight = if (state.value) FontWeight.Bold else FontWeight.Normal,
+            color = if (state.value) Color.Black else Color.Gray
+        )
+    }
+}
```

Comment:
I understand now which part was confusing you in my previous comment, sorry for ambiguity.
I want you to try to remove textDecoration and achieve the underline using Box component. I am giving you the template which should help you implement this.

```
Column(
        modifier = modifier
            .clickable { onItemClick(movieCategoryLabelUiState) }
            .width(Pick right width here so that line is the same size as text),
    ) {
        Text(
            ...
        )

        Box(
            modifier = Modifier
                .padding(top = MaterialTheme.spacing.extraSmall)
                .background(switch colors based on state)
                .add appropirate width (don't hardcode it!)
                .height(add desired height)
        )
    }
```